### PR TITLE
Convert string to UTF-8 before applying hash algorithms

### PR DESCRIPTION
**Pull request by andypaicu**
Fixes https://github.com/w3c/webappsec-csp/issues/109

***
<a href="https://pr-preview.s3.amazonaws.com/andypaicu/webappsec-csp/pull/342.html" title="Last updated on Oct 4, 2018, 9:20 AM GMT (6637dd7)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webappsec-csp/342/f39f3a3...andypaicu:6637dd7.html" title="Last updated on Oct 4, 2018, 9:20 AM GMT (6637dd7)">Diff</a>

### CODE REVIEW

File: index.src.html [hunk 1]
```diff
@@ -4283,8 +4287,8 @@ <h5 id="match-element-to-source-list" algorithm>
   (|type|), and a string (|source|), this algorithm returns "`Matches`" or
   "`Does Not Match`".
 
-  Note: |source| will be interpreted with the encoding of the page in which it
-  is embedded. See the integration points in [[#html-integration]] for more detail.
+  Note: Regardless of the encoding of the page, |source| will be converted to `UTF-8`
```
Comment (annevk) [inline index.src.html]: s/page/document

Comment (annevk) [reply index.src.html]: Though even that might be wrong when it comes to workers. Hmm, maybe it doesn't matter...

File: index.src.html [hunk 2]
```diff
@@ -4314,7 +4318,11 @@ <h5 id="match-element-to-source-list" algorithm>
   5.  If |type| is "`script`" or "`style`", or |unsafe-hashes flag| is
       `true`:
 
-      1.  For each |expression| in |list|:
+      1.  Let |source| be the result of executing <a>UTF-8 encode</a>
+          on the result of executing <a>javascript-string-convert</a>
+          on |source|.
```
Comment (annevk) [inline index.src.html]: You cannot modify a variable with "let". I'd suggest:

> Set _source_ to the result of running UTF-8 encode on the result of converting _source_.

File: index.src.html [hunk 3]
```diff
@@ -140,6 +140,10 @@ <h1>Content Security Policy Level 3</h1>
   type: dfn
     for: WorkerGlobalScope
       text: owner set; url: concept-WorkerGlobalScope-owner-set
+
+spec: INFRA; urlPrefix: https://infra.spec.whatwg.org/
+  type: dfn
+    text: javascript-string-convert; url: javascript-string-convert
```
Comment (annevk) [inline index.src.html]: This shouldn't be needed, the term is exported.

**Comment by andypaicu** [comment]
Thank you @annevk.

I just did it because otherwise it would just say `the result of executing convert` which seemed quite vague without actually clicking on the link to see what sort of conversion you are doing.

**Pull request merged**